polaris/evaluate/utils.py

```diff
@@ -0,0 +1,59 @@
+import pandas as pd
+import numpy as np
+
+from polaris.utils.context import tmp_attribute_change
+from polaris.evaluate import BenchmarkResults, ResultsType
+
+def evaluate_benchmark(model, y_pred, test):
```

Comment:
This is copied, unmodified, from the implementation previously used in the `BenchmarkSpecification` `evaluate` method.

Reply:
Let's start a thread here to continue the conversations we've been having over Slack to decide if/how/when we should modify this utility to support both standard and competition benchmarks. The main issue here is that we want to share all evaluation logic between the two types of benchmarks, but the amount of data available/needed in the competition evaluation service is much less than the client. With the current utility function, we need to hack around this and create dummy data to bypass Pydantic schema checks and get this function to work correctly.

A couple options that come to mind:
- Extend this function to accept both pydantic objects or `dict`/`array` arguments (for standard and competition benchmarks, respectively). We can have type checks at the start of the function which will check the type of the argument passed and then extract all necessary data points according to the type (e.g. for pydantic objects `model.target_cols` and a regular dict `model["target_cols"]`). This should side step the need to change surrounding code which uses this function. 
- Update this function to expect `dict`/`array` arguments only. This will require some changes to the existing client code that evaluates standard benchmarks. However, it will standardize what is expected during evaluation which I like. 

Personally, I think option 2 is the cleanest, and if the changes to existing client code aren't huge, I would prefer to do this as part of the competition feature development. I also am ok with option 1. 

Curious to hear our thoughts on this and other options as well.

Reply:
I think you're right and it wouldn't be very difficult to implement this in a more "functional" (data-in/data-out) kind of way.. it might make sense to just do in this PR, assuming there are no other potentially conflicting changes in progress to benchmark evaluation upstream.